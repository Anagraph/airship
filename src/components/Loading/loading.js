import React from 'react';
import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';

const rotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

const dash = keyframes`
  0% {
    stroke-dasharray: 1, 150;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -124;
  }
`;

const StyledLoading = styled.div`
  height: ${(props) => `${props.size}px`};
  width: ${(props) => `${props.size}px`};

  svg {
    animation: ${rotate} 2s linear infinite;
  }

  circle {
    stroke: ${(props) => {
      console.log(props);
      return !!props.negative
        ? 'rgba(255, 255, 255, 1)'
        : 'rgba(44, 44, 44, 1)';
    }};
    stroke-linecap: round;
    animation: ${dash} 1.5s ease-in-out infinite;
    stroke-width: 4px;
  }
`;

const Loading = ({ size = 16, negative = false }) => {
  return (
    <StyledLoading size={size} negative={negative}>
      <svg viewBox="0 0 50 50">
        <circle cx="25" cy="25" r="20" fill="none" />
      </svg>
    </StyledLoading>
  );
};

Loading.propTypes = {
  size: PropTypes.oneOf([16, 32, 48, 64]),
  negative: PropTypes.bool
};

export default Loading;
