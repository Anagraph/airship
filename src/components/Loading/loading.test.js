import React from 'react';
import Loading from './loading';
import renderer from 'react-test-renderer';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<Loading />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size', () => {
    const component = renderer.create(<Loading size={32} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with negative', () => {
    const component = renderer.create(<Loading negative />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
