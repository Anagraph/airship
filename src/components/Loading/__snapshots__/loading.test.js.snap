// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders with negative 1`] = `
.c0 {
  height: 16px;
  width: 16px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0 svg {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
}

.c0 circle {
  stroke: rgba(255,255,255,1);
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
  stroke-width: 4px;
}

<div
  className="c0"
  size={16}
>
  <svg
    viewBox="0 0 50 50"
  >
    <circle
      cx="25"
      cy="25"
      fill="none"
      r="20"
    />
  </svg>
</div>
`;

exports[`render renders with size 1`] = `
.c0 {
  height: 32px;
  width: 32px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0 svg {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
}

.c0 circle {
  stroke: rgba(44,44,44,1);
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
  stroke-width: 4px;
}

<div
  className="c0"
  size={32}
>
  <svg
    viewBox="0 0 50 50"
  >
    <circle
      cx="25"
      cy="25"
      fill="none"
      r="20"
    />
  </svg>
</div>
`;

exports[`render renders without crashing 1`] = `
.c0 {
  height: 16px;
  width: 16px;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.c0 svg {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
}

.c0 circle {
  stroke: rgba(44,44,44,1);
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
  stroke-width: 4px;
}

<div
  className="c0"
  size={16}
>
  <svg
    viewBox="0 0 50 50"
  >
    <circle
      cx="25"
      cy="25"
      fill="none"
      r="20"
    />
  </svg>
</div>
`;
